src/index/update.rs

```diff
@@ -226,9 +226,13 @@ impl IndexUpdater {
 			let artwork_path = if tags.has_artwork {
 				Some(file_path_string.to_owned())
 			} else {
-				artwork.as_ref().cloned()
+				directory_artwork.as_ref().cloned()
 			};
 
+			if directory_artwork.is_none() {
+				directory_artwork = artwork_path.as_ref().cloned();
```

Comment:
This does solve the previous issue where directories would end up with no artwork at all, but it creates a new problem where artwork from a song may propagate partially to songs in the directory. Consider a directory with no Folder image, and the following content:

- Song 1 (no embedded art)
- Song 2 (embedded art)
- Song 3  (no embedded art)

Assuming the songs get processed in this order, we would end up with art for songs 2 and 3 but not for song 1. I think we should add a preliminary loop when `directory_artwork` is `None` in order to pick a `directory_artwork` that can be applied consistently to all songs (unless they have their own embedded artwork).